### PR TITLE
Add functionality to render a show All button

### PR DESCRIPTION
**Pull request by ghost**
Render a definite number of collections initially with a show All button 
Render all collections as soon as the button is clicked and hide the button when all collections are rendered.
On branch content-library-modal
Changes to be committed:
	modified:   .eslintrc.json
	new file:   listeners/actions/fetch-all-collections-action.js
	modified:   listeners/actions/index.js
	modified:   services/strapi/index.js
	modified:   user-interface/activity-collection/index.js
	modified:   user-interface/app-home/unauthorised-user-view.js
	modified:   user-interface/modals/content-library-modal.js
	
Appropriate comments have been provided at required places.

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -12,6 +12,7 @@
     "object-curly-newline": "off",
     "camelcase": "off",
     "no-console": "off",
-    "eqeqeq": "off"
+    "eqeqeq": "off",
+    "operator-linebreak": "off"
```
Comment (disha1202) [inline .eslintrc.json]: Remove operator line break

File: services/strapi/index.js [hunk 2]
```diff
@@ -54,9 +54,13 @@ async function getSurveyQuestions() {
   }
 }
 
-async function getActivityCollections() {
+async function getActivityCollections(params) {
   try {
-    const url = `${baseUrl}/activity-collections?populate[thumbnail][populate]=*&populate[activities][populate]=*`;
+    // If no endpoint is provided(endpoint is empty), The initial collection is rendered through
```
Comment (disha1202) [inline services/strapi/index.js]: Update the comment

File: services/strapi/index.js [hunk 3]
```diff
@@ -69,9 +73,28 @@ async function getActivityCollections() {
   }
 }
 
+async function getActivityCollection(collectionId) {
+  const collectionQueryParam = 'populate[thumbnail][populate]=*&populate[activities][populate]=*';
+
+  try {
+    const url =
+      `${baseUrl}/activity-collections/${collectionId}?${collectionQueryParam}`;
```
Comment (disha1202) [inline services/strapi/index.js]: Move in a single line

File: services/strapi/index.js [hunk 4]
```diff
@@ -54,9 +54,13 @@ async function getSurveyQuestions() {
...
 
-async function getActivityCollections() {
+async function getActivityCollections(params) {
   try {
-    const url = `${baseUrl}/activity-collections?populate[thumbnail][populate]=*&populate[activities][populate]=*`;
+    // If no endpoint is provided(endpoint is empty), The initial collection is rendered through
+    // intialEndpoint.
+    const initialEndpoint = '/activity-collections?populate=*&pagination[start]=0&pagination[limit]=3';
+    const showAllCollectionEndpoint = 'activity-collections?populate[thumbnail][populate]=*';
+    const url = params?.showAll ? `${baseUrl}/${showAllCollectionEndpoint}` : `${baseUrl}${initialEndpoint}`;
```
Comment (disha1202) [inline services/strapi/index.js]:
```suggestion
    const showAllCollectionEndpoint = '/activity-collections?populate[thumbnail][populate]=*';
    const url = `${baseUrl}/${params?.showAll ? showAllCollectionEndpoint : initialEndpoint}`;
```

**Review by disha1202 — CHANGES_REQUESTED** [review]

File: .eslintrc.json (hunk 1 above)
Comment (ghost) [reply .eslintrc.json]: Done!

File: services/strapi/index.js (hunk 2 above)
Comment (ghost) [reply services/strapi/index.js]: Done!

File: services/strapi/index.js (hunk 3 above)
Comment (ghost) [reply services/strapi/index.js]: Done!

File: services/strapi/index.js (hunk 4 above)
Comment (ghost) [reply services/strapi/index.js]: Done!

File: user-interface/modals/content-library-modal.js [hunk 5]
```diff
@@ -20,16 +19,9 @@ async function getContentLibraryModal(body) {
...
-          text: `*${activity.title}*`,
-          emoji: true,
-        },
-        title_url: 'https://www.youtube.com/embed/RRxQQxiM7AA?feature=oembed&autoplay=1',
-        video_url: 'https://www.youtube.com/embed/RRxQQxiM7AA?feature=oembed&autoplay=1',
-        alt_text: 'How to use Slack?',
-        thumbnail_url: `${baseUrl}${selectedCollection.attributes.thumbnail.data.attributes.url}`,
+        type: 'image',
+        image_url: `${baseUrl}${selectedCollection.attributes.thumbnail.data.attributes.url}`,
+        alt_text: 'inspiration',
```
Comment (disha1202) [inline user-interface/modals/content-library-modal.js]: Should we make this alt text dynamic?

**Review by disha1202 — CHANGES_REQUESTED** [review]

**Review by disha1202 — APPROVED** [review]

**Pull request merged**